cluster-autoscaler/cloudprovider/azure/azure_agent_pool.go

```diff
@@ -211,18 +211,24 @@ func (as *AgentPool) TargetSize() (int, error) {
 	return int(size), nil
 }
 
-func (as *AgentPool) getAllSucceededAndFailedDeployments() (succeededAndFailedDeployments []resources.DeploymentExtended, err error) {
+func (as *AgentPool) getAllSucceededAndFailedDeployments() ([]resources.DeploymentExtended, error) {
 	ctx, cancel := getContextWithCancel()
 	defer cancel()
 
-	deploymentsFilter := "provisioningState eq 'Succeeded' or provisioningState eq 'Failed'"
-	succeededAndFailedDeployments, err = as.manager.azClient.deploymentsClient.List(ctx, as.manager.config.ResourceGroup, deploymentsFilter, nil)
+	allDeployments, err := as.manager.azureClients.deploymentClient.List(ctx, as.manager.config.ResourceGroup)
 	if err != nil {
-		klog.Errorf("getAllSucceededAndFailedDeployments: failed to list succeeded or failed deployments with error: %v", err)
-		return nil, err
+		klog.Errorf("getAllSucceededAndFailedDeployments: failed to list deployments with error: %v", err)
+		return nil, err.Error()
+	}
+
+	result := make([]resources.DeploymentExtended, 0)
+	for _, deployment := range allDeployments {
```

Comment:
Could have small performance impact from client-side filtering (not sure if the original filter is server-side). 
Should be neglectable unless I am missing something. Plus VMAS pool is not properly supported by AKS anymore (will be discussed separately)